### PR TITLE
Fix #5450, don't crash on integer dates in R

### LINKED ISSUE #5450: `IDate` class (an experimental class within R's data.table package) crashes duckdb's dbWriteTable method
### What happens?

https://youtu.be/tfQs7WbVse8?t=84

### To Reproduce

	library(DBI)
	x <- data.frame(data.table::as.IDate("2001-01-01"))
	con <- dbConnect( duckdb::duckdb() , dbdir = tempfile() )
	dbWriteTable( con , "x" , x )

### OS:

windows 10 x64

### DuckDB Version:

0.5.1

### DuckDB Client:

R

### Full Name:

Anthony Damico

### Affiliation:

independent consultant

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

File: tools/rpkg/src/scan.cpp [hunk 1]
```diff
@@ -166,10 +166,19 @@ static unique_ptr<FunctionData> DataFrameScanBind(ClientContext &context, TableF
 			coldata_ptr = (data_ptr_t)INTEGER_POINTER(coldata);
 			break;
 		case RType::DATE:
-		case RType::DATE_INTEGER:
+			if (!IS_NUMERIC(coldata)) {
+				cpp11::stop("DATE_INTEGER should really be integer");
```
Comment (hawkfish) [inline tools/rpkg/src/scan.cpp:170]: "DATE should really be integer" would help narrow down failures.

File: tools/rpkg/src/types.cpp [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
 
 RType RApiTypes::DetectRType(SEXP v, bool integer64) {
-	if (TYPEOF(v) == REALSXP && Rf_inherits(v, "POSIXct")) {
+	if (IS_NUMERIC(v) && Rf_inherits(v, "POSIXct")) {
 		return RType::TIMESTAMP;
-	} else if (TYPEOF(v) == REALSXP && Rf_inherits(v, "Date")) {
+	} else if (IS_NUMERIC(v) && Rf_inherits(v, "Date")) {
 		return RType::DATE;
-	} else if (TYPEOF(v) == INTSXP && Rf_inherits(v, "Date")) {
+	} else if (IS_INTEGER(v) && Rf_inherits(v, "Date")) {
```
Comment (hawkfish) [inline tools/rpkg/src/types.cpp:16]: Shouldn't integral tests come before numeric tests? Or are integers not numeric?

Comment (hannes) [reply tools/rpkg/src/types.cpp:16]: They are not

File: tools/rpkg/src/scan.cpp (hunk 1 above)
Comment (hannes) [reply tools/rpkg/src/scan.cpp:170]: There is an oversight here, the check should be IS_INTEGER

**Pull request merged**